evalml/utils/woodwork_utils.py

```diff
@@ -0,0 +1,112 @@
+                                 pd.StringDtype: 'object'}
+    nullable_to_numpy_mapping_nan = {pd.Int64Dtype: 'float64',
+                                     pd.BooleanDtype: 'object',
+                                     pd.StringDtype: 'object'}
+
+    if isinstance(pd_data, pd.api.extensions.ExtensionArray):
+        if pd.isna(pd_data).any():
+            return pd.Series(pd_data.to_numpy(na_value=np.nan), dtype=nullable_to_numpy_mapping_nan[type(pd_data.dtype)])
+        return pd.Series(pd_data.to_numpy(na_value=np.nan), dtype=nullable_to_numpy_mapping[type(pd_data.dtype)])
+    if (isinstance(pd_data, pd.Series) and type(pd_data.dtype) in nullable_to_numpy_mapping):
+        if pd.isna(pd_data).any():
+            return pd.Series(pd_data.to_numpy(na_value=np.nan), dtype=nullable_to_numpy_mapping_nan[type(pd_data.dtype)], index=pd_data.index, name=pd_data.name)
+        return pd.Series(pd_data.to_numpy(na_value=np.nan), dtype=nullable_to_numpy_mapping[type(pd_data.dtype)], index=pd_data.index, name=pd_data.name)
+    if isinstance(pd_data, pd.DataFrame):
+        for col_name, col in pd_data.iteritems():
+            if type(col.dtype) in nullable_to_numpy_mapping:
+                if pd.isna(pd_data[col_name]).any():
+                    pd_data[col_name] = pd.Series(pd_data[col_name].to_numpy(na_value=np.nan), dtype=nullable_to_numpy_mapping_nan[type(pd_data[col_name].dtype)])
+                else:
+                    pd_data[col_name] = pd_data[col_name].astype(nullable_to_numpy_mapping[type(col.dtype)])
+    return pd_data
+
+
+def _retain_custom_types_and_initalize_woodwork(old_datatable, new_dataframe, ltypes_to_ignore=None):
+    """
+    Helper method which will take an old Woodwork DataTable and a new pandas DataFrame and return a
+    new DataTable that will try to retain as many logical types from the old DataTable that exist in the new
+    pandas DataFrame as possible.
+
+    Arguments:
+        old_datatable (ww.DataTable): Woodwork DataTable to use
+        new_dataframe (pd.DataFrame): Pandas data structure
+        ltypes_to_ignore (list): List of Woodwork logical types to ignore. Columns from the old DataTable that have a logical type
+        specified in this list will not have their logical types carried over to the new DataTable returned
+
+    Returns:
+        A new DataTable where any of the columns that exist in the old input DataTable and the new DataFrame try to retain
+        the original logical type, if possible and not specified to be ignored.
+    """
+    retained_logical_types = {}
+    if ltypes_to_ignore is None:
+        ltypes_to_ignore = []
+    col_intersection = set(old_datatable.columns).intersection(set(new_dataframe.columns))
+    logical_types = old_datatable.logical_types
+    for col in col_intersection:
+        if logical_types[col] in ltypes_to_ignore:
+            continue
+        try:
+            new_dataframe[col].astype(logical_types[col].pandas_dtype)
```

Comment:
Would you be able to skip this check if `new_dataframe[col].dtype == logical_types[col].pandas_dtype`?